core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -50,6 +52,55 @@ public static ObjectMapper mapper() {
     return MAPPER;
   }
 
+  @FunctionalInterface
+  public interface ToJson {
+    void generate(JsonGenerator gen) throws IOException;
+  }
+
+  /**
+   * Helper for writing JSON with a JsonGenerator.
+   *
+   * @param toJson a function to produce JSON using a JsonGenerator
+   * @param pretty whether to pretty-print JSON for readability
+   * @return a JSON string produced from the generator
+   */
+  public static String generate(ToJson toJson, boolean pretty) {
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
```

Comment:
The JsonGenerator is closable, so we should probably be using TWR or closing with finally.